Web_Development/FrontEnd/CSS_Box_Sizing_Models/Readme.md

```diff
@@ -0,0 +1,158 @@
+
+We all know that all elements on a webpage is represnted as rectangular box. Below image represents a element with its various elements.
+
+![image](https://user-images.githubusercontent.com/55577276/134778382-9124b5b1-bec4-4152-b55a-5d9c2aa4920f.png)
+
+*Figure 1 (Reference : https://zellwk.com/blog/understanding-css-box-sizing/)*
+
+
+
+Let’s try understand what each of these properties signifies:
+
+- **Content:** This is the main part of the box where the content put by the user appears. 
+
+- **Padding:** This space is around the content inside an element (transparent).
+
+- **Border:** This  is self explanatory. It usually goes around the padding and content.
+
+- **Margin:** This the space around the box (also transparent).
+
+  
+
+You should remember that total space occupied by any element (box) on the DOM is the combined space occupied by these four properties put together.
+
+
+
+Size of figure 1  box is determined as described below: 
+
+> **Total width of the box-model** = width assigned to the box + padding(right and left) + border(right and left) + margin(right and left)  
+>
+> **Total height of the box-model** = height assigned to the box + padding(top and bottom) + border(top and bottom) + margin(top and bottom) 
+
+
+
+
+
+## Need of using the box-sizing property?
+
+
+![image](https://user-images.githubusercontent.com/55577276/134778925-463dd780-2a4d-41fb-8911-e1c47bc1f578.png)
+
+*Figure 2 (Reference : https://stackoverflow.com/questions/11070071/css-width-in-percentage-and-borders)*
+
+
+
+Now we need to understand Figure 2 better, so let’s consider an example: 
+
+Suppose we have a random element in the dom that is assigned a width of 300px, the element will have the exact width on the DOM, only if there is no padding or margin assigned to the element. If there are any padding and/or border assigned to it then the actual values for the rendered element will differ and be calculated as follows:
+
+> Actual rendered width = initial width assigned to the box + padding(right and left) + border(right and left)  
+>
+> Actual rendered height = initail height assigned to the box + padding(top and bottom) + border(top and bottom)
+
+
+
+Thus, so as to urge the specified width and height as laid out in the planning for the element that's rendered on the DOM, we frequently recalculate the width and height by subtracting the padding and border, either manually or by using the calc(width-valuex) function, (where valuex is that the sum of the padding and/or border assigned thereto element) from their respective dimensions.
+
+
+
+Here the actual width of the element is 300px, but since we assign border and padding to it, we manually recalculate the width as
+
+```css
+my-div{
+  width: 350px;
+  border: solid 5px #82B366;
+  padding: 20px;
+}
+```
+
+width = 300–40(padding-right and padding-left)–10(border-right and border-left) or we can also use the CSSS calc function.
```

Comment:
@onlykingKD Add this inside a code block/as an inline code.

Reply:
donee